recipes/dtc/meta.yaml

```diff
@@ -0,0 +1,122 @@
+{% set version = '1.6.1' %}
+{% set major_ver = version.split(".")[0] %}
+
+package:
+  name: dtc-package
+  version: {{ version }}
+
+source:
+  url: https://git.kernel.org/pub/scm/utils/dtc/dtc.git/snapshot/dtc-{{ version }}.tar.gz
+  sha256: 6401c9a0f577a270df4632bf0f3e5454ccc7a5ca3caefa67a3e1c29c9c6b8c60
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - bison
+    - flex
+    - make
+    - swig
+    - pkg-config
+  host:
+    # Python bindings
+    - python {{ python }}
+    # Optional yaml support
+    - yaml
+  run:
+    - python {{ python }}
+
+outputs:
+  - name: dtc
+    script: install.sh   # [not win]
+    build:
+      missing_dso_whitelist:
+        - /lib64/libc.so.6
```

Comment:
I also don't remember having to specify `/lib64/libc.so.6` in the missing_dso_whitelist before.  I searched the other feedstocks and found several that do this.  Does this mean that I'm directly linking to the system libc instead of the one in the compiler sysroot?  I'm rereading https://github.com/conda/conda-build/blob/0a6efdcfa02484954e7a3fdf5c2ec4bb3cb6682e/conda_build/post.py.  It's been awhile.

Reply:
Looking through the other recipes for CONDA_BUILD_SYSROOT, I see a bunch of packages adding `-isysroot $CONDA_BUILD_SYSROOT` into CFLAGS CXXFLAGS and LDFLAGS.  I tried doing that and removing the missing_dso_whitelist and I get the errors again.  They look like this:
```
Warning: rpath /home/conda/staged-recipes/build_artifacts/dtc-package_1645834285171/_build_env/lib is outside prefix /home/conda/staged-recipes/build_artifacts/dtc-package_1645834285171/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold (removing it) for /home/conda/staged-recipes/build_artifacts/dtc-package_1645834285171/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/libfdt-1.6.1.so
WARNING :: Failed to get_static_lib_exports(/home/conda/staged-recipes/build_artifacts/dtc-package_1645834285171/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/libfdt.a)
Unknown format
  ERROR (libfdt,lib/libfdt-1.6.1.so): /lib64/libc.so.6 not found in packages, sysroot(s) nor the missing_dso_whitelist.
.. is this binary repackaging?
```

@conda-forge/help-c-cpp what's the deal with libc and the missing_dso_whitelist?  At one point, I thought that we expected stuff to link against the sysroot libc at build time and then at runtime link the system libc.  Should I just leave the missing_dso_whitelist in my recipe or do I need to modify the build so that it links differently?

Reply:
I think I got this figured out after looking more at https://github.com/conda/conda-build/blob/77463386355f3a726437311d3fa1f39a59afc2e9/conda_build/post.py#L1062-L1072, I realized that the split packages need to list the compiler in the `requirements.build` section of their output so that the sysroot will be found in build_prefix and libc will be found to be okay by the overlinking logic.  Testing this locally and will push an update to remove the missing_dso_whitelist and also remove the `-isysroot` arguments because I'm pretty sure that those are baked into the compilers correctly for linux.  Perhaps they are needed for osx though?